### PR TITLE
Bump upper bound on the time package.

### PR DESCRIPTION
**Pull request by Fuuzetsu**
This compiles fine and allows us to build test-framework with recent
GHC HEAD.

### CODE REVIEW

File: core/test-framework.cabal [hunk 1]
```diff
@@ -52,7 +52,7 @@ Library
                               , containers     >= 0.1    && < 0.7
                               , regex-posix    >= 0.72   && < 0.96
                               , old-locale     >= 1.0    && < 1.1
-                              , time           >= 1.1.2  && < 1.9
+                              , time           >= 1.1.2  && < 2
```
Comment (hvr) [inline core/test-framework.cabal]: This is *not* a PVP compliant upper bound. See https://pvp.haskell.org/faq/#semver for more details.

**Comment by hvr** [comment]
And as far as GHC HEAD is concerned, we've implemented a rather rich `allow-newer`-syntax, see https://cabal.readthedocs.io/en/latest/nix-local-build.html#cfg-field-allow-newer for this very purpose so you don't have to wait for package maintainers to vet new major versions for compability if you are confident that things are sound.

**Comment by Fuuzetsu** [comment]
>And as far as GHC HEAD is concerned, we've implemented a rather rich allow-newer-syntax, see https://cabal.readthedocs.io/en/latest/nix-local-build.html#cfg-field-allow-newer for this very purpose so you don't have to wait for package maintainers to vet new major versions for compability if you are confident that things are sound.

I'm using nix (not cabal-install nor stack) and I do have an override that allows the build to proceed regardless of versions, I just think it's better to make changes upstream instead of accumulating local cruft.

**Comment by nh2** [comment]
The commit was updated, is this good to go?

**Comment by hvr** [comment]
yes, thanks!

**Pull request merged**